src/django_idom/websocket_consumer.py

```diff
@@ -23,14 +25,15 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
 
     async def connect(self) -> None:
         await super().connect()
-        self._idom_dispatcher_future = asyncio.ensure_future(self._run_dispatch_loop())
+        self._idom_dispatcher_thread = Thread(
+            target=asyncio.run,
+            args=(self._run_dispatch_loop(),),
+        )
+        self._idom_dispatcher_thread.daemon = True
+        self._idom_dispatcher_thread.start()
 
     async def disconnect(self, code: int) -> None:
-        if self._idom_dispatcher_future.done():
-            await self._idom_dispatcher_future
-        else:
-            self._idom_dispatcher_future.cancel()
-        await super().disconnect(code)
+        self._idom_dispatcher_thread.join(timeout=0)
```

Comment:
Implementation mentioned above results in `RuntimeError: Cannot close a running event loop`. 
Some quick googling suggested I add `self._dispatch_loop.call_soon_threadsafe(self._dispatch_loop.stop)` before running close, however, that did not resolve it.

Another google search suggested this was a bug with threads in `asyncio.run`, so I attempted to replace it with `asyncio.ensure_future` and pass in a `loop` argument. Same results.

Reply:
I've implemented a version that closes the thread through inducing an exception within `dispatch_single_view`, since I'm unsure how to resolve this within a threaded asyncio event loop.